tvb_epilepsy/custom/readers_custom.py

```diff
@@ -78,65 +79,64 @@ def read_t1(self, h5_path):
-        sensors_dict = {}
-        for sensor_file in ensure_list(sensor_files):
-            sensor = self.read_sensors(os.path.join(root_folder, sensor_file[0]), s_type)
-            if isinstance(sensor, Sensors):
-                projection = []
-                if len(sensor_file) > 1:
-                    projection_file = os.path.join(root_folder, sensor_file[1])
-                    if os.path.isfile(projection_file):
-                        projection = self.read_projection(os.path.join(root_folder, sensor_file[1]), s_type)
-                if projection == []:
-                    warning("Calculating projection matrix based solely on euclidean distance!")
-                    projection = sensor.calculate_projection(conn)
-                sensors_dict[sensor] = projection
-        return sensors_dict
+    def read_gain_matrix(self, path, s_type):
+        if os.path.isfile(path):
+            return np.load(path)
+        else:
+            warning("\nNo Projection Matrix file found at path " + path + "!")
+            return []
 
     def read_head(self, root_folder, name='',
                   connectivity_file="Connectivity.h5",
                   surface_file="CorticalSurface.h5",
                   region_mapping_file="RegionMapping.h5",
                   volume_mapping_file="VolumeMapping.h5",
                   structural_mri_file="StructuralMRI.h5",
-                  seeg_sensors_files=[("SensorsSEEG_114.h5", ""), ("SensorsSEEG_125.h5", "")],
-                  eeg_sensors_files=[("eeg_brainstorm_65.txt", "projection_eeg_65_surface_16k.npy")],
-                  meg_sensors_files=[("meg_brainstorm_276.txt", "projection_meg_276_surface_16k.npy")],
+                  seeg_sensors_files=[("SensorsSEEG_114.h5", ), ("SensorsSEEG_125.h5", )],
+                  eeg_sensors_files=[],
+                  meg_sensors_files=[],
                   ):
-
         conn = self.read_connectivity(os.path.join(root_folder, "Connectivity.h5"))
         srf = self.read_cortical_surface(os.path.join(root_folder, "CorticalSurface.h5"))
         rm = self.read_region_mapping(os.path.join(root_folder, "RegionMapping.h5"))
         vm = self.read_volume_mapping(os.path.join(root_folder, "VolumeMapping.h5"))
         t1 = self.read_volume_mapping(os.path.join(root_folder, "StructuralMRI.h5"))
-
-        seeg_sensors_dict = self.read_sensors_projections(root_folder, conn, seeg_sensors_files, Sensors.TYPE_SEEG)
-
-        eeg_sensors_dict = self.read_sensors_projections(root_folder, conn, eeg_sensors_files, Sensors.TYPE_EEG)
-
-        meg_sensors_dict = self.read_sensors_projections(root_folder, conn, meg_sensors_files, Sensors.TYPE_MEG)
-
-        return Head(conn, srf, rm, vm, t1, name, eeg_sensors_dict, meg_sensors_dict, seeg_sensors_dict)
+        sensorsSEEG = []
+        for s_files in ensure_list(seeg_sensors_files):
+            sensorsSEEG.append(self.read_sensors(s_files, root_folder, Sensors.TYPE_SEEG))
+        sensorsEEG = []
```

Comment:
Sensors could be read by scanning the Head folder.

Reply:
Nice. But are the filenames settled?